app/logic/volunteers.py

```diff
@@ -19,45 +21,67 @@ def getEventLengthInHours(startTime, endTime, eventDate):
+            eventParticipant = EventParticipant.get_or_none(user=userObject, event=participantData['event'])
+            if userObject:
                 try:
-                    if participantData['checkbox_'+ username] == "on": #if the user is marked as present
-                        (EventParticipant.update({EventParticipant.hoursEarned: float(participantData['inputHours_'+ username]),EventParticipant.attended: True}).where(EventParticipant.event == participantData['event'],
-                                                 EventParticipant.user == participantData[f'username{user}'])).execute()
-
+                    if participantData[f'checkbox_{username}']: #if the user is marked as present
+                        hoursEarned = float(participantData['inputHours_'+ username])
+                        if eventParticipant:
+                            ((EventParticipant
+                                .update({
+                                    EventParticipant.hoursEarned: hoursEarned})
+                                .where(
+                                    EventParticipant.event==participantData['event'],
+                                    EventParticipant.user==participantData[f'username{user}']))
+                                .execute())
+                        else:
+                            (EventParticipant
+                                .create(
+                                    user=user,
+                                    event=event,
+                                    hoursEarned=hoursEarned))
                 except (KeyError):   #if there is no checkbox for user then they are not present for the event.
-                    (EventParticipant.update({EventParticipant.attended: False, EventParticipant.hoursEarned: 0}).where(EventParticipant.event == participantData['event'],
-                        EventParticipant.user == participantData[f'username{user}'])).execute()
-                except Exception as e:
-                    print(e)
-                    return False
+                    if eventParticipant:
+                        ((EventParticipant
+                            .update({
+                                EventParticipant.hoursEarned: 0})
+                            .where(
+                                EventParticipant.event==participantData['event'],
+                                EventParticipant.user==participantData[f'username{user}']))
+                            .execute())
+                    else:
+                        (EventParticipant
+                            .create(
+                                user=user,
+                                event=event,
+                                hoursEarned=0))
             else:
                 return False
         else:
             break
     return True
 
 
-def addVolunteerToEvent(user, volunteerEventID, eventLengthInHours):
+def addVolunteerToEvent(user, volunteerEventID):
```

Comment:
We should clarify this method name now that the semantics have changed. It's not clear from the name whether it is messing with participant or rsvp

Reply:
Also, it seems like this change should have broken a test, but there isn't a corresponding test change. If it's missing a test, you should add it.